core/src/main/java/org/opensearch/sql/utils/JsonUtils.java

```diff
@@ -79,6 +85,58 @@ public static ExprValue castJson(ExprValue json) {
     return processJsonNode(jsonNode);
   }
 
+  /**
+   * Extract value of JSON string at given JSON path.
+   *
+   * @param json JSON string (e.g. "{\"hello\": \"world\"}").
+   * @param paths list of JSON path (e.g. "$.hello")
+   * @return ExprValue of value at given path of json string.
+   */
+  public static ExprValue extractJson(ExprValue json, ExprValue... paths) {
+    List<ExprValue> resultList = new ArrayList<>();
+
+    for (ExprValue path : paths) {
+      System.out.println("Processing path: " + path);
+      if (json == LITERAL_NULL || json == LITERAL_MISSING) {
+        return json;
+      }
```

Comment:
Probably don't want to use `==` for comparison (my understanding is that this will compare the memory locations of the objects, and so depends on `ExprNullValue` and `ExprMissingValue` being implemented as singletons.

```suggestion
      if (json.isNull() || json.isMissing()) {
        return json;
      }
```